backend/src/main.rs

```diff
@@ -46,10 +51,17 @@ fn init_tracer(_log_file: &Option<PathBuf>, trace_level: Level) {
 }
 
 fn router(cors: CorsLayer) -> Router {
-    // Add api
-    Router::new()
-        // Unsecured Routes
-        .route("/", get(root))
+    let mut frontend = env::current_exe().expect("Couldn't get current executable path.");
+    frontend.pop();
+    frontend.push(FRONTEND_FOLDER);
+    tracing::info!("serving frontend at: {frontend:?}");
+    let mut router = Router::new()
+        // Frontend
+        .nest_service("/", ServeDir::new(frontend));
+
+    // Add API
+    router = new_api_route!(router, "/root", get(root)).expect("Couldn't register new API route");
```

Comment:
Now that you have a check for OpenAPI compliance, add a test that calls `router` function. This test allow to validate, that the declaration and actual API matches

Reply:
UPDATE: Now instead of macro call, `ApiRouterExt` trait was added. It's `api_route` method utilizes `axum`'s `Router` functions and generic types alongside with `utopia`'s OpenAPI documentation providing API validation. 9 different checks were added for possible scenarios (might add more in the future)

Reply:
Anyway, the test in `main.rs` that calls `router` function is needed. This test should validate that our Api is correctly registered. Reason: we should see the problem when running the tests within GitHub actions

Reply:
Tests start [here](https://github.com/qoollo/bob-management/pull/10/files#diff-46ebab5b973eb7487aa51b201c0292920632f5d75171d31720213d550f754ba5R272). I don't think putting them in `main.rs` is a good idea tho

Reply:
But these tests does not verify that all API methods are properly registered in OpenAPI specification

Reply:
Ah, I get it now what do you want. Will do